appengine/ndb/entities/snippets.py

```diff
@@ -0,0 +1,258 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from google.appengine.ext import ndb
+
+
+class Account(ndb.Model):
+    username = ndb.StringProperty()
+    userid = ndb.IntegerProperty()
+    email = ndb.StringProperty()
+
+
+def create_sandy_with_keywords():
+    sandy = Account(username='Sandy',
+                    userid=123,
+                    email='sandy@gmail.com')
+    return sandy
+
+
+def create_sandy_manually():
+    sandy = Account()
+    sandy.username = 'Sandy'
+    sandy.userid = 123
+    sandy.email = 'sandy@gmail.com'
+    return sandy
+
+
+def create_with_type_error_in_constructor():
+    bad = Account(username='Sandy',
+                  userid='not integer')  # raises an exception
+    return bad
+
+
+def assign_with_type_error(sandy):
+    sandy.username = 42  # raises an exception
+
+
+def store_sandy(sandy):
+    sandy_key = sandy.put()
+    return sandy_key
+
+
+def url_safe_sandy_key(sandy_key):
+    url_string = sandy_key.urlsafe()
+    return url_string
+
+
+def get_sandy_from_urlsafe(url_string):
+    sandy_key = ndb.Key(urlsafe=url_string)
+    sandy = sandy_key.get()
+    return sandy
+
+
+def id_from_urlsafe(url_string):
+    key = ndb.Key(urlsafe=url_string)
+    kind_string = key.kind()
+    ident = key.id()
+    return key, ident, kind_string
+
+
+def update_key(key):
+    sandy = key.get()
+    sandy.email = 'sandy@gmail.co.uk'
+    sandy.put()
+
+
+def delete_sandy(sandy):
+    sandy.key.delete()
+
+
+def create_sandy_named_key():
+    account = Account(username='Sandy', userid=1234, email='some@where.com',
+                      id='SOME@WHERE.COM')
```

Comment:
please use `@example.com`, because that domain is specifically reserved for examples.